### PR TITLE
Unified UI

### PR DESCRIPTION
**Pull request by stas-vilchik**

### CODE REVIEW

File: server/sonar-web/src/main/js/apps/quality-gates/components/Condition.js [hunk 1]
```diff
@@ -42,19 +42,19 @@ export default class Condition extends Component {
   componentDidMount() {
     const { condition } = this.props;
 
-    if (!condition.id) {
-      this.refs.operator.focus();
+    // TODO looks like `this.opetator` is always null or undefined
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/quality-gates/components/Condition.js]: INFO Complete the task associated to this TODO comment. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS1135)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: server/sonar-web/src/main/js/apps/coding-rules/templates/rule/coding-rules-rule-profile.hbs [hunk 2]
```diff
@@ -55,32 +55,28 @@
   {{#if actions.edit}}
     {{#unless isBuiltIn}}
       <td class="coding-rules-detail-quality-profile-actions">
-        <div class="button-group">
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/apps/coding-rules/templates/rule/coding-rules-rule-profile.hbs]: I noticed that in some case we have a border under the buttons and sometimes not : 
image

Another one : 
image

File: server/sonar-web/src/main/js/components/icons-components/OrganizationIcon.tsx [hunk 3]
```diff
@@ -17,25 +17,22 @@
...
-      viewBox="0 0 16 16">
+      height={size}
+      viewBox="0 0 16 16"
+      version="1.1"
+      xmlnsXlink="http://www.w3.org/1999/xlink"
+      xmlSpace="preserve">
       <path
-        style={{ fill: theme.blue }}
+        style={{ fill }}
         d="M13.5 6c-.4 0-.7.1-1.1.2L11 4.8v-.3C11 3.1 9.9 2 8.5 2S6 3.1 6 4.5v.2L4.5 6.2c-.3-.1-.7-.2-1-.2C2.1 6 1 7.1 1 8.5S2.1 11 3.5 11 6 9.9 6 8.5c0-.7-.3-1.3-.7-1.7l1-1c.4.6 1 1 1.7 1.1V9c-1.1.2-2 1.2-2 2.4C6 12.9 7.1 14 8.5 14s2.5-1.1 2.5-2.5c0-1.2-.9-2.2-2-2.4V6.9c.7-.1 1.2-.5 1.6-1.1l1 1c-.4.4-.6 1-.6 1.6 0 1.4 1.1 2.5 2.5 2.5s2.5-1 2.5-2.4S14.9 6 13.5 6zm-10 4C2.7 10 2 9.3 2 8.5S2.7 7 3.5 7 5 7.7 5 8.5 4.3 10 3.5 10zm6.5 1.5c0 .8-.7 1.5-1.5 1.5S7 12.3 7 11.5 7.7 10 8.5 10s1.5.7 1.5 1.5zM8.5 6C7.7 6 7 5.3 7 4.5S7.7 3 8.5 3s1.5.7 1.5 1.5S9.3 6 8.5 6zm5 4c-.8 0-1.5-.7-1.5-1.5S12.7 7 13.5 7s1.5.7 1.5 1.5-.7 1.5-1.5 1.5z"
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/icons-components/OrganizationIcon.tsx]: MAJOR Exceeds maximum line length of 180 [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS103)

File: server/sonar-web/src/main/js/components/icons-components/ListIcon.tsx [hunk 4]
```diff
@@ -17,24 +17,21 @@
...
-      viewBox="0 0 16 16">
+      height={size}
+      viewBox="0 0 16 16"
+      version="1.1"
+      xmlnsXlink="http://www.w3.org/1999/xlink"
+      xmlSpace="preserve">
       <path
-        fill="currentColor"
+        style={{ fill }}
         d="M15.045 11.526v1.007q0 0.204-0.149 0.354t-0.354 0.149h-13.084q-0.204 0-0.354-0.149t-0.149-0.354v-1.006q0-0.204 0.149-0.354t0.354-0.149h13.084q0.204 0 0.354 0.149t0.149 0.354zM15.045 8.506v1.006q0 0.204-0.149 0.354t-0.354 0.149h-13.084q-0.204 0-0.354-0.149t-0.149-0.354v-1.006q0-0.204 0.149-0.354t0.354-0.149h13.084q0.204 0 0.354 0.149t0.149 0.354zM15.045 5.487v1.006q0 0.204-0.149 0.354t-0.354 0.149h-13.084q-0.204 0-0.354-0.149t-0.149-0.354v-1.006q0-0.204 0.149-0.354t0.354-0.149h13.084q0.204 0 0.354 0.149t0.149 0.354zM15.045 2.468v1.006q0 0.204-0.149 0.354t-0.354 0.149h-13.084q-0.204 0-0.354-0.149t-0.149-0.354v-1.006q0-0.204 0.149-0.354t0.354-0.149h13.084q0.204 0 0.354 0.149t0.149 0.354z"
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/icons-components/ListIcon.tsx]: MAJOR Exceeds maximum line length of 180 [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS103)

File: server/sonar-web/src/main/js/components/icons-components/SortAscIcon.tsx [hunk 5]
```diff
@@ -17,24 +17,21 @@
...
+      height={size}
+      viewBox="0 0 16 16"
+      version="1.1"
+      xmlnsXlink="http://www.w3.org/1999/xlink"
+      xmlSpace="preserve">
       <path
-        style={{ fill: 'currentColor' }}
+        style={{ fill }}
         d="M6.571 12.857q0 0.107-0.089 0.214l-2.848 2.848q-0.089 0.080-0.205 0.080-0.107 0-0.205-0.080l-2.857-2.857q-0.134-0.143-0.063-0.313 0.071-0.179 0.268-0.179h1.714v-12.286q0-0.125 0.080-0.205t0.205-0.080h1.714q0.125 0 0.205 0.080t0.080 0.205v12.286h1.714q0.125 0 0.205 0.080t0.080 0.205zM16 14v1.714q0 0.125-0.080 0.205t-0.205 0.080h-7.429q-0.125 0-0.205-0.080t-0.080-0.205v-1.714q0-0.125 0.080-0.205t0.205-0.080h7.429q0.125 0 0.205 0.080t0.080 0.205zM14.286 9.429v1.714q0 0.125-0.080 0.205t-0.205 0.080h-5.714q-0.125 0-0.205-0.080t-0.080-0.205v-1.714q0-0.125 0.080-0.205t0.205-0.080h5.714q0.125 0 0.205 0.080t0.080 0.205zM12.571 4.857v1.714q0 0.125-0.080 0.205t-0.205 0.080h-4q-0.125 0-0.205-0.080t-0.080-0.205v-1.714q0-0.125 0.080-0.205t0.205-0.080h4q0.125 0 0.205 0.080t0.080 0.205zM10.857 0.286v1.714q0 0.125-0.080 0.205t-0.205 0.080h-2.286q-0.125 0-0.205-0.080t-0.080-0.205v-1.714q0-0.125 0.080-0.205t0.205-0.080h2.286q0.125 0 0.205 0.080t0.080 0.205z"
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/icons-components/SortAscIcon.tsx]: MAJOR Exceeds maximum line length of 180 [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS103)

File: server/sonar-web/src/main/js/components/icons-components/SortDescIcon.tsx [hunk 6]
```diff
@@ -17,24 +17,21 @@
...
+      height={size}
+      viewBox="0 0 16 16"
+      version="1.1"
+      xmlnsXlink="http://www.w3.org/1999/xlink"
+      xmlSpace="preserve">
       <path
-        style={{ fill: 'currentColor' }}
+        style={{ fill }}
         d="M10.857 14v1.714q0 0.125-0.080 0.205t-0.205 0.080h-2.286q-0.125 0-0.205-0.080t-0.080-0.205v-1.714q0-0.125 0.080-0.205t0.205-0.080h2.286q0.125 0 0.205 0.080t0.080 0.205zM6.571 12.857q0 0.107-0.089 0.214l-2.848 2.848q-0.089 0.080-0.205 0.080-0.107 0-0.205-0.080l-2.857-2.857q-0.134-0.143-0.063-0.313 0.071-0.179 0.268-0.179h1.714v-12.286q0-0.125 0.080-0.205t0.205-0.080h1.714q0.125 0 0.205 0.080t0.080 0.205v12.286h1.714q0.125 0 0.205 0.080t0.080 0.205zM12.571 9.429v1.714q0 0.125-0.080 0.205t-0.205 0.080h-4q-0.125 0-0.205-0.080t-0.080-0.205v-1.714q0-0.125 0.080-0.205t0.205-0.080h4q0.125 0 0.205 0.080t0.080 0.205zM14.286 4.857v1.714q0 0.125-0.080 0.205t-0.205 0.080h-5.714q-0.125 0-0.205-0.080t-0.080-0.205v-1.714q0-0.125 0.080-0.205t0.205-0.080h5.714q0.125 0 0.205 0.080t0.080 0.205zM16 0.286v1.714q0 0.125-0.080 0.205t-0.205 0.080h-7.429q-0.125 0-0.205-0.080t-0.080-0.205v-1.714q0-0.125 0.080-0.205t0.205-0.080h7.429q0.125 0 0.205 0.080t0.080 0.205z"
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/icons-components/SortDescIcon.tsx]: MAJOR Exceeds maximum line length of 180 [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS103)

File: server/sonar-web/src/main/js/components/icons-components/TreeIcon.tsx [hunk 7]
```diff
@@ -17,22 +17,21 @@
...
-      viewBox="0 0 16 16">
+      height={size}
+      viewBox="0 0 16 16"
+      version="1.1"
+      xmlnsXlink="http://www.w3.org/1999/xlink"
+      xmlSpace="preserve">
       <path
-        fill="currentColor"
+        style={{ fill }}
         d="M15.045 2.467c0-0.277-0.225-0.503-0.503-0.503h-13.084c-0.277 0-0.503 0.225-0.503 0.503v1.007c0 0.277 0.225 0.503 0.503 0.503h13.084c0.277 0 0.503-0.225 0.503-0.503v-1.007zM15.045 5.487c0-0.277-0.194-0.503-0.432-0.503h-11.216c-0.238 0-0.432 0.225-0.432 0.503v1.007c0 0.277 0.193 0.503 0.432 0.503h11.216c0.238 0 0.432-0.225 0.432-0.503v-1.007zM15.045 8.506c0-0.277-0.161-0.503-0.359-0.503h-9.346c-0.198 0-0.359 0.225-0.359 0.503v1.007c0 0.277 0.161 0.503 0.359 0.503h9.346c0.198 0 0.359-0.225 0.359-0.503v-1.007zM15.045 11.527c0-0.277-0.129-0.503-0.287-0.503h-7.477c-0.159 0-0.288 0.225-0.288 0.503v1.007c0 0.277 0.129 0.503 0.288 0.503h7.477c0.159 0 0.287-0.225 0.287-0.503v-1.007z"
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/icons-components/TreeIcon.tsx]: MAJOR Exceeds maximum line length of 180 [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 5 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 5 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: server/sonar-web/src/main/js/apps/background-tasks/components/Search.js [hunk 8]
```diff
@@ -54,9 +55,9 @@ export default class Search extends React.PureComponent {
     this.props.onFilterUpdate(date);
   }
 
-  handleQueryChange(query /*: string */) {
+  handleQueryChange = (query /*: string */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/background-tasks/components/Search.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/components/ui/buttons.tsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    event.preventDefault();
+    event.currentTarget.blur();
+    event.stopPropagation();
+    if (this.props.onClick) {
+      this.props.onClick();
+    }
+  };
+
+  render() {
+    const { children, className, color = theme.darkBlue, onClick, ...props } = this.props;
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/ui/buttons.tsx]: MAJOR Remove this useless assignment to local variable "onClick". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1854)

File: server/sonar-web/src/main/js/components/common/MultiSelect.js [hunk 10]
```diff
@@ -104,8 +106,8 @@ export default class MultiSelect extends React.PureComponent {
     }
   };
 
-  handleSearchChange = ({ target } /*: { target: HTMLInputElement } */) => {
-    this.onSearchQuery(this.props.validateSearchInput(target.value));
+  handleSearchChange = (value /*: string */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/common/MultiSelect.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/components/controls/SearchBox.tsx [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+            color={theme.gray60}
+            onClick={this.handleResetClick}>
+            <ClearIcon size={12} />
+          </ButtonIcon>
+        )}
+
+        {tooShort && (
+          <span
+            className="search-box-note"
+            title={translateWithParameters('select2.tooShort', minLength!)}>
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/controls/SearchBox.tsx]: MINOR Forbidden non null assertion [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS2966)

File: server/sonar-web/src/main/js/components/controls/SearchBox.tsx [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+            {translateWithParameters('select2.tooShort', minLength!)}
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/controls/SearchBox.tsx]: MINOR Forbidden non null assertion [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS2966)

File: server/sonar-web/src/main/js/app/components/search/Search.js [hunk 13]
```diff
@@ -297,10 +306,18 @@ export default class Search extends React.PureComponent {
     this.setState({ selected });
   };
 
+  handleClick = (event /*: Event */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/app/components/search/Search.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/app/components/search/Search.js [hunk 14]
```diff
@@ -297,10 +306,18 @@ export default class Search extends React.PureComponent {
...
+    event.stopPropagation();
+  };
+
   innerRef = (component /*: string */, node /*: HTMLElement */) => {
     this.nodes[component] = node;
   };
 
+  searchInputRef = (node /*: HTMLInputElement | null */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/app/components/search/Search.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/components/icons-components/SearchIcon.tsx [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+      className={className}
+      width={size}
+      height={size}
+      viewBox="0 0 16 16"
+      version="1.1"
+      xmlnsXlink="http://www.w3.org/1999/xlink"
+      xmlSpace="preserve">
+      <path
+        style={{ fill }}
+        d="M10.308 7.077c0-.89-.316-1.65-.949-2.283a3.111 3.111 0 0 0-2.282-.948c-.89 0-1.65.316-2.283.948a3.111 3.111 0 0 0-.948 2.283c0 .89.316 1.65.948 2.282a3.111 3.111 0 0 0 2.283.949c.89 0 1.65-.316 2.282-.949a3.111 3.111 0 0 0 .949-2.282zm3.692 6c0 .25-.091.466-.274.649a.887.887 0 0 1-.65.274.857.857 0 0 1-.648-.274L9.954 11.26c-.86.596-1.82.894-2.877.894a4.989 4.989 0 0 1-1.972-.4 5.076 5.076 0 0 1-1.623-1.082A5.076 5.076 0 0 1 2.4 9.049 4.989 4.989 0 0 1 2 7.077c0-.688.133-1.345.4-1.972a5.076 5.076 0 0 1 1.082-1.623A5.076 5.076 0 0 1 5.105 2.4 4.989 4.989 0 0 1 7.077 2c.687 0 1.345.133 1.972.4a5.076 5.076 0 0 1 1.623 1.082c.454.454.815.995 1.082 1.623.266.627.4 1.284.4 1.972a4.938 4.938 0 0 1-.894 2.877l2.473 2.474a.883.883 0 0 1 .267.649z"
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/icons-components/SearchIcon.tsx]: MAJOR Exceeds maximum line length of 180 [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 13 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**